logstash-core/lib/logstash/runner.rb

```diff
@@ -184,6 +184,11 @@ def run(args)
   end
 
   def execute
+    # Only when execute is have the CLI options been added to the @settings
+    # We invoke post_process to apply extra logic to them.
+    # The post_process callbacks have been added in environment.rb
+    @settings.post_process
```

Comment:
@ph well, I'd rather wait till we have to deal with that complexity TBH. I'm not anticipating this being a common thing to do *yet*.